### PR TITLE
Ns 4

### PR DESCRIPTION
**Pull request by Poostaq**
Working third screen + some refactor done.

### CODE REVIEW

**Review by Malynna — COMMENTED** [review]
approved after changing comments

File: game/CharacterSheet/CharacterSheet.gd [hunk 1]
```diff
@@ -140,22 +118,21 @@ func _return_modifier_value_or_n(value):
...
-	match attribute:
-		0:	self.character_stats_element.agi_modifiers["EthnicityAttributeModifier"] = 1
-		1:	self.character_stats_element.per_modifiers["EthnicityAttributeModifier"] = 1
-		2:	self.character_stats_element.cha_modifiers["EthnicityAttributeModifier"] = 1
-		3:	self.character_stats_element.wit_modifiers["EthnicityAttributeModifier"] = 1
-		4:	self.character_stats_element.bod_modifiers["EthnicityAttributeModifier"] = 1
-		_:	print("DIDNT MATCH ANYTHING")
+	if attribute != null or (attribute <= 4 and attribute >= 0):
+		self.character_stats_element.attribute_modifiers_dicts[attribute]["EthnicityAttributeModifier"] = 1
+		print("MODIFIED")
```
[the remark this answers is not in the document]
Comment (Poostaq) [reply game/CharacterSheet/CharacterSheet.gd]: deleted

File: game/AttributesPage/AttributesStep.gd [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+		var final_value = ceil((result_rolls[0]+result_rolls[1]+result_rolls[2])/3.0)
+		math_list[x].bbcode_text = ("[center] (%s+%s+%s)/3= " % result_rolls) + \
+		"%s" % str(final_value)
+		roll_list[x].bbcode_text = "[center]%s" % str(final_value)
+
+
+func _roll_attribute_above_six():
+	var average = 0
+	var results = []
+	while average <= 6:
```
[the remark this answers is not in the document]
Comment (Poostaq) [reply game/AttributesPage/AttributesStep.gd]: fixed

File: game/CharacterSheet/CharacterSheet.gd [hunk 3]
```diff
@@ -140,22 +118,21 @@ func _return_modifier_value_or_n(value):
...
-	self.character_stats_element.per_modifiers["EthnicityAttributeModifier"] = 0
-	self.character_stats_element.cha_modifiers["EthnicityAttributeModifier"] = 0
-	self.character_stats_element.wit_modifiers["EthnicityAttributeModifier"] = 0
-	self.character_stats_element.bod_modifiers["EthnicityAttributeModifier"] = 0
+	for element in self.character_stats_element.attribute_modifiers_dicts:
+		element["EthnicityAttributeModifier"] = 0
 
+func clear_base_rolls_attributes():
+	for element in self.character_stats_element.attribute_modifiers_dicts:
+		element["BaseRoll"] = 0
```
[the remark this answers is not in the document]
Comment (Poostaq) [reply game/CharacterSheet/CharacterSheet.gd:128]: added line

File: game/CharacterSheet/CharacterSheet.gd [hunk 4]
```diff
@@ -140,22 +118,21 @@ func _return_modifier_value_or_n(value):
 
 
 func _clear_bonus_attribute():
-	self.character_stats_element.agi_modifiers["EthnicityAttributeModifier"] = 0
-	self.character_stats_element.per_modifiers["EthnicityAttributeModifier"] = 0
-	self.character_stats_element.cha_modifiers["EthnicityAttributeModifier"] = 0
-	self.character_stats_element.wit_modifiers["EthnicityAttributeModifier"] = 0
-	self.character_stats_element.bod_modifiers["EthnicityAttributeModifier"] = 0
+	for element in self.character_stats_element.attribute_modifiers_dicts:
+		element["EthnicityAttributeModifier"] = 0
```
[the remark this answers is not in the document]
Comment (Poostaq) [reply game/CharacterSheet/CharacterSheet.gd:123]: added line

File: game/CharacterSheet/CharacterSheet.gd [hunk 5]
```diff
@@ -140,22 +118,21 @@ func _return_modifier_value_or_n(value):
...
-		3:	self.character_stats_element.wit_modifiers["EthnicityAttributeModifier"] = 1
-		4:	self.character_stats_element.bod_modifiers["EthnicityAttributeModifier"] = 1
-		_:	print("DIDNT MATCH ANYTHING")
+	if attribute != null or (attribute <= 4 and attribute >= 0):
+		self.character_stats_element.attribute_modifiers_dicts[attribute]["EthnicityAttributeModifier"] = 1
+		print("MODIFIED")
 
+func _set_base_roll(attribute=null, value=0):
+	if attribute != null or (attribute <= 5 and attribute >= 0):
+		self.character_stats_element.attribute_modifiers_dicts[attribute]["BaseRoll"] = value
```
[the remark this answers is not in the document]
Comment (Poostaq) [reply game/CharacterSheet/CharacterSheet.gd:138]: added line

**Review by Malynna — APPROVED** [review]

**Pull request merged**